packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2643,39 +2651,59 @@ export class ContainerRuntime
 			}
 			const batchStartCsn = processResult.batchStartCsn;
 			const batch = processResult.messages;
-			const messages: {
+			const processedMessages: {
 				message: InboundSequencedContainerRuntimeMessage;
 				localOpMetadata: unknown;
 			}[] = local
 				? this.pendingStateManager.processPendingLocalBatch(batch, batchStartCsn)
 				: batch.map((message) => ({ message, localOpMetadata: undefined }));
-			messages.forEach(({ message, localOpMetadata }) => {
+			processedMessages.forEach(({ message, localOpMetadata }) => {
 				const msg: MessageWithContext = {
 					message,
 					local,
-					modernRuntimeMessage,
+					isRuntimeMessage: true,
 					savedOp,
 					localOpMetadata,
 				};
-				this.ensureNoDataModelChanges(() => this.processCore(msg));
+				this.ensureNoDataModelChanges(() => this.processRuntimeMessage(msg));
 			});
 		} else {
-			const msg: MessageWithContext = {
-				message: messageCopy as InboundSequencedContainerRuntimeMessageOrSystemMessage,
-				local,
-				modernRuntimeMessage,
-				savedOp,
-			};
-			this.ensureNoDataModelChanges(() => this.processCore(msg));
+			// Check if message.type is one of values in ContainerMessageType
+			// eslint-disable-next-line import/no-deprecated
```

Comment:
Weird... the deprecation says to use the one in driver-utils... but that one just checks for the "op" envelope.

We should probably un-deprecate this one and maybe rename it to "isUnpackedRuntimeMessage", in this PR or a follow-up.

Reply:
This utility is exported from container-runtime package, so I don't think I can change it until the next breaking change version release. Created an ADO https://dev.azure.com/fluidframework/internal/_workitems/edit/5704/?workitem=10256

Reply:
Can you add comment with the ADO item here? We should be very careful adding uses of deprecated APIs. We either need to use a non-deprecated version or undeprecate.